# 03b – Finaler Durchlauf nach Bugfix: beam=12, len=20 über alle Sessions

## Kontext: Der Bugfix

In `segmentation.py` wurde ein Fehler behoben: der Fallback-Wert für
`min_duration_off_frames` las fälschlicherweise `min_duration_on` statt `min_duration_off`.
Dieses Notebook wiederholt den finalen Durchlauf aus `03_` mit dem **korrigierten Code**.

**Zweck:** Reiner Sanity-Check – dieser Lauf zeigt wie hoch die WER nach dem Bugfix
auf allen 25 Dev-Sessions ist, bevor der Grid-Search beginnt. Der Output
`output_final_bugfix_bs12_len20` wird von keinem anderen Notebook als Input genutzt.
Der Grid-Search in `02k_` läuft unabhängig per Monkey-Patch direkt auf 5 Sessions.

## Unterschied zu `03_`

| Aspekt | `03_` | `03b_` |
|--------|-------|--------|
| Datenpfad | `dev_without_central_videos/dev` | `dev` (vollständig) |
| Bugfix | Nein | Ja |
| Output-Prefix | `output_final_bs12_len20` | `output_final_bugfix_bs12_len20` |
| Ergebnis-CSV | `final_results.csv` | `final_bugfix_results.csv` |

## 1 – GPU-Check

In [1]:
!nvidia-smi --query-gpu=index,memory.total,memory.used,memory.free --format=csv,noheader

0, 81920 MiB, 14989 MiB, 66048 MiB
1, 81920 MiB, 4 MiB, 81033 MiB
2, 81920 MiB, 4 MiB, 81033 MiB
3, 81920 MiB, 4 MiB, 81033 MiB


In [2]:
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "2"  # macht "unsere" GPU zur 0 im Prozess

import torch
print("n_gpu:", torch.cuda.device_count())
print("current device:", torch.cuda.current_device())
print("device name:", torch.cuda.get_device_name(0))


n_gpu: 1
current device: 0
device name: NVIDIA A100-SXM4-80GB


## 2 – Pfad-Konfiguration

In [3]:
project_baseline_path = "/home/josch080/Projektgruppe/mcorec_baseline" # anpassen
session = "*" # alle Sessions oder einzelne session_132
session_dir = f"data-bin/dev/{session}"

In [4]:
import os, sys
os.chdir(project_baseline_path)
print("CWD:", os.getcwd())

CWD: /home/josch080/Projektgruppe/mcorec_baseline


In [5]:
from script import inference

/home/josch080/Projektgruppe/mcorec_baseline/script/inference.py:22: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.
  if not hasattr(np, "object"):


## 3 – Inference-Parameter

In [6]:
model_type = "avsr_cocktail"
checkpoint_path = "model-bin/avsr_cocktail_mcorec_finetune" # BL4-Checkpoint
beam_size = 12
max_length = 20
output_dir_name = f"output_final_bugfix_bs12_len20" # Bugfix-spezifischer Output-Name

## 4 – Inference starten

In [7]:
sys.argv = [
    "notebook",
    "--model_type", model_type,
    "--session_dir", session_dir,
    "--checkpoint_path", checkpoint_path,
    "--beam_size", str(beam_size),
    "--max_length", str(max_length), 
    "--output_dir_name", output_dir_name,

    ]
inference.main()

Loading avsr_cocktail model...


/home/josch080/Projektgruppe/mcorec_train/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 25 sessions using avsr_cocktail model
Processing session session_44


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:23<06:56, 83.39s/it]


Processing speakers:  33%|███▎      | 2/6 [02:02<03:50, 57.60s/it]


Processing speakers:  50%|█████     | 3/6 [03:02<02:54, 58.27s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 19/19 [00:20<00:00,  1.09s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:36<01:37, 48.90s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:24<00:48, 48.52s/it]


Processing speaker spk_5 track 1 of 3: 100%|██████████| 23/23 [00:23<00:00,  1.00s/it]

Processing speaker spk_5 track 2 of 3: 100%|██████████| 2/2 [00:00<00:00,  2.00it/s]

Processing speakers: 100%|██████████| 6/6 [04:54<00:00, 49.07s/it]


Processing session session_57


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 25/25 [01:13<00:00,  2.96s/it]

Processing speakers:  20%|██        | 1/5 [01:35<06:20, 95.16s/it]


Processing speakers:  40%|████      | 2/5 [02:17<03:11, 63.93s/it]


Processing speakers:  60%|██████    | 3/5 [02:49<01:39, 49.65s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 23/23 [00:25<00:00,  1.10s/it]

Processing speakers:  80%|████████  | 4/5 [03:42<00:50, 50.69s/it]


Processing speakers: 100%|██████████| 5/5 [05:00<00:00, 60.02s/it]


Processing session session_139


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:17<06:26, 77.26s/it]


Processing speakers:  33%|███▎      | 2/6 [02:25<04:48, 72.20s/it]


Processing speakers:  50%|█████     | 3/6 [03:13<03:02, 60.90s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 21/21 [00:23<00:00,  1.14s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:47<01:40, 50.32s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:31<00:47, 47.94s/it]


Processing speakers: 100%|██████████| 6/6 [05:26<00:00, 54.40s/it]


Processing session session_133


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 29/29 [00:41<00:00,  1.44s/it]

Processing speakers:  17%|█▋        | 1/6 [00:52<04:22, 52.48s/it]


Processing speaker spk_1 track 1 of 2: 100%|██████████| 18/18 [00:28<00:00,  1.56s/it]

Processing speakers:  33%|███▎      | 2/6 [01:51<03:45, 56.29s/it]


Processing speakers:  50%|█████     | 3/6 [03:13<03:24, 68.24s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:49<01:50, 55.27s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:50<00:57, 57.24s/it]


Processing speakers: 100%|██████████| 6/6 [06:18<00:00, 63.14s/it]


Processing session session_134


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:17<06:27, 77.44s/it]


Processing speakers:  33%|███▎      | 2/6 [02:20<04:36, 69.04s/it]


Processing speakers:  50%|█████     | 3/6 [03:17<03:09, 63.30s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:28<02:12, 66.47s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 21/21 [00:19<00:00,  1.10it/s]

Processing speakers:  83%|████████▎ | 5/6 [05:16<00:59, 59.88s/it]


Processing speakers: 100%|██████████| 6/6 [06:11<00:00, 61.91s/it]


Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:35<02:57, 35.46s/it]


Processing speakers:  33%|███▎      | 2/6 [01:21<02:46, 41.68s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 20/20 [00:20<00:00,  1.03s/it]

Processing speakers:  50%|█████     | 3/6 [01:50<01:47, 35.90s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 17/17 [00:20<00:00,  1.22s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 13/13 [00:19<00:00,  1.53s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:50<01:30, 45.41s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:16<00:38, 38.50s/it]


Processing speakers: 100%|██████████| 6/6 [04:16<00:00, 42.75s/it]


Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:15<01:17, 15.48s/it]


Processing speakers:  33%|███▎      | 2/6 [00:43<01:31, 22.93s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.47it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.77it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.76it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:16<00:00,  5.36s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:09<00:00,  4.79s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 9/9 [00:16<00:00,  1.88s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:04<00:00,  1.39s/it]

Processing speakers:  50%|█████     | 3/6 [01:47<02:04, 41.58s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:32<01:26, 43.14s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:32<00:48, 48.97s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 25/25 [00:39<00:00,  1.59s/it]

Processing speakers: 100%|██████████| 6/6 [04:28<00:00, 44.71s/it]


Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 30/30 [01:08<00:00,  2.30s/it]

Processing speakers:  17%|█▋        | 1/6 [01:30<07:30, 90.17s/it]


Processing speakers:  33%|███▎      | 2/6 [02:16<04:18, 64.57s/it]


Processing speakers:  50%|█████     | 3/6 [03:19<03:11, 63.93s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:35<02:16, 68.35s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 17/17 [00:22<00:00,  1.35s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:25<01:02, 62.01s/it]


Processing speakers: 100%|██████████| 6/6 [06:10<00:00, 61.78s/it]


Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 31/31 [00:41<00:00,  1.35s/it]

Processing speakers:  20%|██        | 1/5 [00:57<03:49, 57.28s/it]


Processing speakers:  40%|████      | 2/5 [02:14<03:26, 68.83s/it]


Processing speakers:  60%|██████    | 3/5 [03:07<02:03, 61.77s/it]


Processing speakers:  80%|████████  | 4/5 [04:28<01:09, 69.48s/it]


Processing speakers: 100%|██████████| 5/5 [05:12<00:00, 62.48s/it]


Processing session session_140


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:48<04:04, 48.91s/it]


Processing speakers:  33%|███▎      | 2/6 [01:27<02:52, 43.08s/it]


Processing speakers:  50%|█████     | 3/6 [02:46<02:57, 59.11s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 5/5 [00:07<00:00,  1.41s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 14/14 [00:10<00:00,  1.31it/s]

Processing speakers:  67%|██████▋   | 4/6 [03:23<01:40, 50.48s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:06<00:47, 48.00s/it]


Processing speaker spk_5 track 1 of 3: 100%|██████████| 7/7 [00:06<00:00,  1.10it/s]

Processing speaker spk_5 track 2 of 3: 100%|██████████| 6/6 [00:04<00:00,  1.39it/s]

Processing speakers: 100%|██████████| 6/6 [04:39<00:00, 46.60s/it]


Processing session session_137


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:32<02:42, 32.49s/it]


Processing speakers:  33%|███▎      | 2/6 [02:12<04:47, 71.93s/it]


Processing speakers:  50%|█████     | 3/6 [02:47<02:46, 55.42s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 23/23 [00:19<00:00,  1.18it/s]

Processing speakers:  67%|██████▋   | 4/6 [03:33<01:43, 51.64s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:24<00:51, 51.46s/it]


Processing speakers: 100%|██████████| 6/6 [05:15<00:00, 52.53s/it]


Processing session session_53


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speakers:  20%|██        | 1/5 [01:14<04:56, 74.11s/it]


Processing speakers:  40%|████      | 2/5 [02:13<03:16, 65.40s/it]


Processing speakers:  60%|██████    | 3/5 [03:05<01:58, 59.22s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 23/23 [01:03<00:00,  2.77s/it]

Processing speakers:  80%|████████  | 4/5 [04:27<01:08, 68.40s/it]


Processing speakers: 100%|██████████| 5/5 [05:09<00:00, 61.99s/it]


Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:41<03:28, 41.75s/it]


Processing speakers:  33%|███▎      | 2/6 [02:08<04:33, 68.50s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 16/16 [00:44<00:00,  2.78s/it]

Processing speakers:  50%|█████     | 3/6 [03:38<03:54, 78.31s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 19/19 [00:43<00:00,  2.32s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:48<02:29, 74.79s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:53<01:11, 71.21s/it]


Processing speakers: 100%|██████████| 6/6 [07:01<00:00, 70.29s/it]


Processing session session_42


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 14/14 [00:13<00:00,  1.06it/s]

Processing speakers:  17%|█▋        | 1/6 [00:48<04:00, 48.01s/it]


Processing speakers:  33%|███▎      | 2/6 [02:18<04:51, 72.89s/it]


Processing speakers:  50%|█████     | 3/6 [03:53<04:09, 83.19s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 4/4 [00:05<00:00,  1.35s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:48<02:24, 72.07s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:49<01:08, 68.07s/it]


Processing speakers: 100%|██████████| 6/6 [06:37<00:00, 66.25s/it]


Processing session session_48


Processing speakers:   0%|          | 0/3 [00:00<?, ?it/s]


Processing speaker spk_1 track 1 of 3: 100%|██████████| 2/2 [00:01<00:00,  1.45it/s]

Processing speaker spk_1 track 2 of 3: 100%|██████████| 6/6 [00:04<00:00,  1.36it/s]

Processing speakers:  33%|███▎      | 1/3 [00:51<01:42, 51.28s/it]


Processing speakers:  67%|██████▋   | 2/3 [01:24<00:40, 40.38s/it]


Processing speaker spk_3 track 1 of 10: 100%|██████████| 4/4 [00:02<00:00,  1.44it/s]

Processing speaker spk_3 track 2 of 10: 0it [00:00, ?it/s]

Processing speaker spk_3 track 3 of 10: 100%|██████████| 2/2 [00:01<00:00,  1.23it/s]

Processing speaker spk_3 track 4 of 10: 100%|██████████| 9/9 [00:17<00:00,  1.91s/it]

Processing speaker spk_3 track 5 of 10: 100%|██████████| 9/9 [00:18<00:00,  2.04s/it]

Processing speaker spk_3 track 6 of 10: 100%|██████████| 2/2 [00:01<00:00,  1.01it/s]

Processing speaker spk_3 track 7 of 10: 100%|██████████| 2/2 [00:06<00:00,  3.02s/it]

Processing speaker spk_3 track 8 of 10: 100%|██████████| 1/1 [00:07<00:00,  7.87s/it]

Processing speaker spk_3 track 9 of 10: 0it [00:00, ?it/s]

Processing speakers: 100%|██████████| 3/3 [02:26<00:00, 49.00s/it]


Processing session session_135


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 16/16 [00:21<00:00,  1.33s/it]

Processing speakers:  20%|██        | 1/5 [00:57<03:50, 57.63s/it]


Processing speakers:  40%|████      | 2/5 [01:58<02:59, 59.81s/it]


Processing speakers:  60%|██████    | 3/5 [03:18<02:17, 68.66s/it]


Processing speakers:  80%|████████  | 4/5 [04:16<01:04, 64.44s/it]


Processing speakers: 100%|██████████| 5/5 [05:23<00:00, 64.65s/it]


Processing session session_51


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:47<03:55, 47.12s/it]


Processing speaker spk_1 track 1 of 2: 100%|██████████| 16/16 [00:26<00:00,  1.67s/it]

Processing speakers:  33%|███▎      | 2/6 [01:48<03:42, 55.58s/it]


Processing speakers:  50%|█████     | 3/6 [02:26<02:22, 47.61s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:57<02:09, 64.53s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:44<00:58, 58.37s/it]


Processing speakers: 100%|██████████| 6/6 [05:17<00:00, 52.88s/it]


Processing session session_56


Processing speakers:   0%|          | 0/4 [00:00<?, ?it/s]


Processing speakers:  25%|██▌       | 1/4 [00:28<01:26, 29.00s/it]


Processing speakers:  50%|█████     | 2/4 [02:05<02:17, 68.82s/it]


Processing speakers:  75%|███████▌  | 3/4 [03:32<01:17, 77.04s/it]


Processing speakers: 100%|██████████| 4/4 [03:47<00:00, 56.80s/it]


Processing session session_132


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:38<08:12, 98.52s/it]


Processing speakers:  33%|███▎      | 2/6 [02:44<05:18, 79.59s/it]


Processing speakers:  50%|█████     | 3/6 [04:16<04:15, 85.26s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:49<02:09, 64.63s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:39<00:59, 59.37s/it]


Processing speaker spk_5 track 1 of 5: 100%|██████████| 13/13 [00:24<00:00,  1.89s/it]

Processing speaker spk_5 track 2 of 5: 100%|██████████| 4/4 [00:02<00:00,  1.47it/s]

Processing speaker spk_5 track 3 of 5: 100%|██████████| 12/12 [00:18<00:00,  1.58s/it]

Processing speaker spk_5 track 4 of 5: 100%|██████████| 4/4 [00:04<00:00,  1.22s/it]

Processing speakers: 100%|██████████| 6/6 [06:32<00:00, 65.37s/it]


Processing session session_138


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:29<02:29, 29.94s/it]


Processing speakers:  33%|███▎      | 2/6 [02:04<04:31, 67.81s/it]


Processing speakers:  50%|█████     | 3/6 [02:48<02:51, 57.25s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 24/24 [00:43<00:00,  1.81s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:47<01:55, 57.83s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:23<00:49, 49.84s/it]


Processing speaker spk_5 track 1 of 4: 100%|██████████| 7/7 [00:07<00:00,  1.04s/it]

Processing speaker spk_5 track 2 of 4: 100%|██████████| 19/19 [00:29<00:00,  1.57s/it]

Processing speaker spk_5 track 3 of 4: 100%|██████████| 7/7 [00:05<00:00,  1.22it/s]

Processing speakers: 100%|██████████| 6/6 [05:09<00:00, 51.51s/it]


Processing session session_41


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 19/19 [00:24<00:00,  1.32s/it]

Processing speakers:  17%|█▋        | 1/6 [00:26<02:12, 26.40s/it]


Processing speaker spk_1 track 1 of 2: 0it [00:00, ?it/s]

Processing speakers:  33%|███▎      | 2/6 [02:04<04:33, 68.32s/it]


Processing speaker spk_2 track 1 of 2: 0it [00:00, ?it/s]

Processing speakers:  50%|█████     | 3/6 [03:27<03:46, 75.40s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 3/3 [00:09<00:00,  3.16s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:20<02:13, 66.51s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:16<01:02, 62.70s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 33/33 [00:54<00:00,  1.65s/it]

Processing speakers: 100%|██████████| 6/6 [06:11<00:00, 61.99s/it]


Processing session session_136


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:24<07:01, 84.38s/it]


Processing speakers:  33%|███▎      | 2/6 [02:16<04:20, 65.20s/it]


Processing speakers:  50%|█████     | 3/6 [03:05<02:54, 58.04s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:04<01:57, 58.53s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 28/28 [01:02<00:00,  2.22s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:19<01:04, 64.16s/it]


Processing speakers: 100%|██████████| 6/6 [05:52<00:00, 58.69s/it]


Processing session session_141


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speakers:  20%|██        | 1/5 [00:44<02:57, 44.31s/it]


Processing speakers:  40%|████      | 2/5 [01:47<02:45, 55.17s/it]


Processing speakers:  60%|██████    | 3/5 [02:37<01:46, 53.07s/it]


Processing speakers:  80%|████████  | 4/5 [03:12<00:45, 45.75s/it]


Processing speakers: 100%|██████████| 5/5 [03:55<00:00, 47.14s/it]


Processing session session_52


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:39<03:19, 39.99s/it]


Processing speakers:  33%|███▎      | 2/6 [00:58<01:48, 27.19s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 2/2 [00:01<00:00,  1.37it/s]

Processing speakers:  50%|█████     | 3/6 [02:28<02:47, 55.86s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:11<01:41, 50.87s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:31<00:39, 39.75s/it]


Processing speakers: 100%|██████████| 6/6 [05:19<00:00, 53.33s/it]


Processing session session_55


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [00:39<00:00,  1.46s/it]

Processing speakers:  20%|██        | 1/5 [01:06<04:27, 66.79s/it]


Processing speakers:  40%|████      | 2/5 [02:18<03:28, 69.60s/it]


Processing speakers:  60%|██████    | 3/5 [03:14<02:06, 63.37s/it]


Processing speakers:  80%|████████  | 4/5 [03:32<00:45, 45.44s/it]


Processing speakers: 100%|██████████| 5/5 [05:22<00:00, 64.46s/it]


## 5 – Evaluation via Subprocess

In [8]:
import subprocess, sys, re, pandas as pd

eval_session_dir = session_dir   
eval_output_prefix = output_dir_name

cmd = [
    sys.executable,
    "script/evaluate.py",
    "--session_dir", eval_session_dir,
    "--output_dir_name", eval_output_prefix,
    "--label_dir_name", "labels",
]

result = subprocess.run(cmd, capture_output=True, text=True)

# Ausgabe trotzdem im Notebook sehen:
print(result.stdout)

# Ausgabe auch in eine Datei schreiben
with open("eval_log_final_results_bugfix.txt", "w") as f:
    f.write(result.stdout)


Evaluating 25 sessions

=== Evaluating session session_44 ===

--- Evaluating output dir: output_final_bugfix_bs12_len20 ---
Conversation clustering F1 score: 0.6
Speaker to WER: {'spk_0': 0.4581, 'spk_1': 0.6388, 'spk_2': 0.5083, 'spk_3': 0.756, 'spk_4': 0.7203, 'spk_5': 0.6301}
Speaker clustering F1 score: {'spk_0': 1.0, 'spk_1': 1.0, 'spk_2': 0.5, 'spk_3': 0.5, 'spk_4': 0.5, 'spk_5': 0.5}
Joint ASR-Clustering Error Rate: {'spk_0': 0.22905, 'spk_1': 0.3194, 'spk_2': 0.50415, 'spk_3': 0.628, 'spk_4': 0.61015, 'spk_5': 0.56505}

=== Evaluating session session_57 ===

--- Evaluating output dir: output_final_bugfix_bs12_len20 ---
Conversation clustering F1 score: 1.0
Speaker to WER: {'spk_0': 0.3433, 'spk_1': 0.3249, 'spk_2': 0.5083, 'spk_3': 0.5451, 'spk_4': 0.5783}
Speaker clustering F1 score: {'spk_0': 1.0, 'spk_1': 1.0, 'spk_2': 1.0, 'spk_3': 1.0, 'spk_4': 1.0}
Joint ASR-Clustering Error Rate: {'spk_0': 0.17165, 'spk_1': 0.16245, 'spk_2': 0.25415, 'spk_3': 0.27255, 'spk_4': 0.28915}


## 6 – Ergebnis-Parsing

In [9]:
pattern = re.compile(
    r"Results for output dir prefix variant '([^']+)':\s+"
    r"\s*Average Conversation Clustering F1 score: ([0-9.]+)\s+"
    r"\s*Average Speaker WER: ([0-9.]+)\s+"
    r"\s*Average Joint ASR-Clustering Error Rate: ([0-9.]+)",
    re.MULTILINE
)

rows = []
for m in pattern.finditer(result.stdout):
    rows.append({
        "exp": m.group(1),                 # z.B. "output_avsr_cocktail_finetuned"
        "avg_conv_f1": float(m.group(2)),
        "avg_speaker_wer": float(m.group(3)),
        "avg_joint_error": float(m.group(4)),
    })

baseline_results = pd.DataFrame(rows)
baseline_results["model"] = baseline_results["exp"].str.replace("^output_", "", regex=True)
baseline_results

exp  avg_conv_f1  avg_speaker_wer  \
0  output_final_bugfix_bs12_len20     0.855282         0.519048   

   avg_joint_error                    model  
0         0.340854  final_bugfix_bs12_len20

In [ ]:
baseline_results.to_csv(
    "final_bugfix_results.csv",
    index=False
)


## 7 – Vergleich: Bugfix-Final vs. BL4-Baseline

In [10]:
import pandas as pd

# ========= 1) Final aus dem Notebook (du hast es "baseline_results" genannt) =========
final_df = baseline_results.copy()   # <- DAS sind deine finalen Werte (bs12/len20)

# ========= 2) Baseline aus CSV laden (Finetuned bs3/len15) =========
base_df = pd.read_csv("results_baseline_dev_without_central_videos.csv")

BASELINE_EXP = "output_avsr_cocktail_finetuned"   # Finetuned Modell (bs3/len15)
# Final-Exp-Name: entweder explizit oder einfach erste Zeile nehmen
FINAL_EXP = "output_final_bugfix_bs12_len20"

# --- robuste Auswahl ---
def get_row(df, exp_name):
    if "exp" not in df.columns:
        raise ValueError(f"Keine Spalte 'exp' gefunden. Spalten: {list(df.columns)}")
    hit = df[df["exp"].astype(str) == exp_name]
    if hit.empty:
        print(f"[WARN] exp='{exp_name}' nicht gefunden. Verfügbare exp: {df['exp'].tolist()}")
        return df.iloc[0]
    return hit.iloc[0]

b = get_row(base_df, BASELINE_EXP)
f = get_row(final_df, FINAL_EXP)

# ========= 3) PPT-taugliche Runs-Tabelle =========
runs = pd.DataFrame([
    {"run": "baseline (finetuned, beam=3,len=15)", "exp": b["exp"],
     "avg_conv_f1": float(b["avg_conv_f1"]),
     "avg_speaker_wer": float(b["avg_speaker_wer"]),
     "avg_joint_error": float(b["avg_joint_error"])},
    {"run": "final (beam=12,len=20)", "exp": f["exp"],
     "avg_conv_f1": float(f["avg_conv_f1"]),
     "avg_speaker_wer": float(f["avg_speaker_wer"]),
     "avg_joint_error": float(f["avg_joint_error"])},
])

display(runs.round(6))

# ========= 4) Vergleich (F1 ↑ besser, WER/JER ↓ besser) =========
comparison = pd.DataFrame([
    {
        "metric": "Avg Conv Clustering F1 (↑)",
        "baseline": runs.loc[0, "avg_conv_f1"],
        "final": runs.loc[1, "avg_conv_f1"],
        "improvement_abs": runs.loc[1, "avg_conv_f1"] - runs.loc[0, "avg_conv_f1"],
    },
    {
        "metric": "Avg Speaker WER (↓)",
        "baseline": runs.loc[0, "avg_speaker_wer"],
        "final": runs.loc[1, "avg_speaker_wer"],
        "improvement_abs": runs.loc[0, "avg_speaker_wer"] - runs.loc[1, "avg_speaker_wer"],
    },
    {
        "metric": "Avg Joint Error (↓)",
        "baseline": runs.loc[0, "avg_joint_error"],
        "final": runs.loc[1, "avg_joint_error"],
        "improvement_abs": runs.loc[0, "avg_joint_error"] - runs.loc[1, "avg_joint_error"],
    },
])
comparison["improvement_rel_%"] = (comparison["improvement_abs"] / comparison["baseline"]) * 100

display(comparison.round(6))


run                             exp  \
0  baseline (finetuned, beam=3,len=15)  output_avsr_cocktail_finetuned   
1               final (beam=12,len=20)  output_final_bugfix_bs12_len20   

   avg_conv_f1  avg_speaker_wer  avg_joint_error  
0     0.815282         0.498727         0.354674  
1     0.855282         0.519048         0.340854

metric  baseline     final  improvement_abs  \
0  Avg Conv Clustering F1 (↑)  0.815282  0.855282         0.040000   
1         Avg Speaker WER (↓)  0.498727  0.519048        -0.020321   
2         Avg Joint Error (↓)  0.354674  0.340854         0.013820   

   improvement_rel_%  
0           4.906278  
1          -4.074544  
2           3.896579

## 8 – Interpretation & Ausblick

Dieser Lauf **dokumentiert** den Bugfix-Effekt isoliert: beam=12, len=20 mit
korrigierter Segmentierung verschlechtert die WER gegenüber dem Pre-Bugfix-Stand.

**Wichtig:** `output_final_bugfix_bs12_len20` wird von keinem anderen Notebook
als Input genutzt. Der Grid-Search in `02k_` läuft unabhängig davon direkt
über den Monkey-Patch auf den 5 Dev-Sessions – nicht auf diesem Output.

`03b_` dient als **Nachweis** dass der Bugfix allein nicht ausreicht und
eine Neu-Kalibrierung der Segmentierungsparameter (`min_duration_on/off`)
notwendig ist – motiviert also `02k_`.

Die finale optimierte Konfiguration (min_on=1.0, min_off=1.2, beam=12, len=20)
folgt in `04_dev_final_results` und `04_eval_final_results`.
